code/datums/weather/weather.dm

```diff
@@ -192,11 +192,11 @@
 /datum/weather/proc/can_weather_act(mob/living/mob_to_check)
 	var/turf/mob_turf = get_turf(mob_to_check)
 
-	if(!(mob_turf.z in impacted_z_levels))
-		if(mob_turf)
-			return
+	if(mob_turf && !(mob_turf.z in impacted_z_levels))
+		return
 
-		if(!(mob_to_check.loc.z in impacted_z_levels)) // in this case, get_turf(mob_to_check) =/= mob_to_check.loc, so we have to proceed with checking the location's type
+	if(!(mob_turf) && mob_to_check.loc)
+		if(!(mob_to_check.loc.z in impacted_z_levels)) // in this case, get_turf(mob_to_check) is null (and therefore mob_to_check is in a non-turf), so we have to proceed with checking the location's type
```

Comment:
get_turf doesn't return a turf if the mob is a turf, it returns a turf if any part of the mob's location is a turf. So if the mob was inside a locker .loc would return the locker, but get_turf would return the turf the locker is on.

If get_turf(mob_to_check) is null, that means the object in question is in nullspace, which things are not often in. It also means that .loc will have a nonsense z level. So this subcheck will just runtime since loc is null,  unless the mob is inside a thing that's also in nullspace, in which case it'll just evaluate to true

looking at the linked issue, the mob that caused the runtime was a dummy, which is a like, ahhhh prop mob we fill with equipment when we want to display something, that's what shows up in the create character menu 
![image](https://user-images.githubusercontent.com/58055496/124524543-5aa9eb80-ddb0-11eb-8f5a-8cd7392b5e4d.png)